server/server.ino

```diff
@@ -98,91 +161,64 @@ unsigned int readUnsignedIntBlocking() {
   return value;
 };
 
+//===================  START RoDI primitives  ===================
 /**
  * RoDI primitives
  */
-
-void executeGetBothIR() {
-  unsigned int sensorLeftState = analogRead(SENSOR_LEFT_PIN);
-  unsigned int sensorRightState = analogRead(SENSOR_RIGHT_PIN);
-  writeInteger(sensorLeftState);
-  writeInteger(sensorRightState);
-};
-
-void executeGetLeftIR() {
-  unsigned int sensorLeftState = analogRead(SENSOR_LEFT_PIN);
-  writeInteger(sensorLeftState);
+void executeTurnLeadOn() {
+  rodiHardware.turnLeadOn();
 };
 
-void executeGetRightIR() {
-  unsigned int sensorRightState = analogRead(SENSOR_RIGHT_PIN);
-  writeInteger(sensorRightState);
+void executeTurnLeadOff() {
+  rodiHardware.turnLeadOff();
 };
 
 void executeGetSonar() {
-  digitalWrite(SONAR_TRIGGER_PIN, LOW);
-  delayMicroseconds(SONAR_SHORT_DELAY);
-
-  digitalWrite(SONAR_TRIGGER_PIN, HIGH);
-  delayMicroseconds(SONAR_LONG_DELAY);
-
-  digitalWrite(SONAR_TRIGGER_PIN, LOW);
-  float sonarDuration = (float) pulseIn(SONAR_ECHO_PIN, HIGH);
-  long sonarDistance = sonarDuration / SONAR_MAGIC_NUMBER;
-
-  if (sonarDistance > SONAR_MAX_DISTANCE) {
-    sonarDistance = SONAR_MAX_DISTANCE;
-  }
-
+  long sonarDistance = rodiHardware.getSonar();
   Serial.write(sonarDistance);
 };
 
-void executeTurnLeadOn() {
-  digitalWrite(LED_PIN, HIGH);
-};
-
-void executeTurnLeadOff() {
-  digitalWrite(LED_PIN, LOW);
-};
-
 void executeMoveLeftServo() {
   char speed = readByteBlocking();
-  isLeftMotorAttached = moveMotor(leftMotor, isLeftMotorAttached, SERVO_LEFT_PIN, speed, 1);
+  isLeftMotorAttached = rodiHardware.moveMotor(leftMotor, isLeftMotorAttached, SERVO_LEFT_PIN, speed, 1);
 };
 
 void executeMoveRightServo() {
   char speed = readByteBlocking();
-  isRightMotorAttached = moveMotor(rightMotor, isRightMotorAttached, SERVO_RIGHT_PIN, speed, -1);
+  isRightMotorAttached = rodiHardware.moveMotor(rightMotor, isRightMotorAttached, SERVO_RIGHT_PIN, speed, -1);
```

Comment:
As suggested by @toriak, it would be nice to have two methods here (`rodiHardware.moveRightServo` and `rodiHardware.moveLeftServo`) that would only take the speed as a parameter. `rodiHardware.moveMotor` should be a private thing.